x/stakeibc/keeper/lsm.go

```diff
@@ -160,24 +160,32 @@ func (k Keeper) GetValidatorFromLSMTokenDenom(denom string, validators []*types.
 
 // Checks if we need to issue an ICQ to check if a validator was slashed
 // The query runs at periodic intervals defined by the ValidatorSlashQueryInterval
-func (k Keeper) ShouldCheckIfValidatorWasSlashed(ctx sdk.Context, validator types.Validator, stakeAmount sdkmath.Int) bool {
+// The interval is represented as percent of TVL
+// (e.g. 1% means every LS that causes the progress to breach 1% of TVL triggers the query)
+func (k Keeper) ShouldCheckIfValidatorWasSlashed(
+	ctx sdk.Context,
+	validator types.Validator,
+	totalHostZoneStake sdkmath.Int,
+	transactionStakeAmount sdkmath.Int,
+) bool {
 	params := k.GetParams(ctx)
-	queryInterval := sdk.NewIntFromUint64(params.ValidatorSlashQueryInterval)
+	queryInterval := sdk.NewDecWithPrec(int64(params.ValidatorSlashQueryInterval), 2) // percentage
+	checkpoint := queryInterval.Mul(sdk.NewDecFromInt(totalHostZoneStake)).TruncateInt()
 
-	// If the query interval is disabled, do not submit a query
-	// This should not be possible with the current parameter validation
-	//  function which enforces that it's greater than 0
-	if queryInterval.IsZero() {
+	// If the checkpoint is zero - that means either the interval parameter is 0
+	// (which should not be possible), or that the total host zone stake is 0
+	// In either case, do not submit the query
+	if checkpoint.IsZero() {
 		return false
 	}
 
 	oldProgress := validator.SlashQueryProgressTracker
-	newProgress := validator.SlashQueryProgressTracker.Add(stakeAmount)
+	newProgress := validator.SlashQueryProgressTracker.Add(transactionStakeAmount)
 
 	// Submit query if the query interval checkpoint has been breached
 	// Ex: Query Interval: 1000, Old Progress: 900, New Progress: 1100
 	//     => OldProgress/Interval: 0, NewProgress/Interval: 1
-	return oldProgress.Quo(queryInterval).LT(newProgress.Quo(queryInterval))
+	return oldProgress.Quo(checkpoint).LT(newProgress.Quo(checkpoint))
```

Comment:
I don't think this algorithm will work exactly the same way as before, because originally queryInterval was a fixed number but now checkpoint changes with each time we call this function.  It is possible that an amount which would have pushed us over the limit has also increased the checkpoint step size enough that a check doesn't happen.

Reply:
Just re-read the way this function gets called, so the total amount on the host zone doesn't get updated until the callback, so this function doing the check at the start isn't going to have the size of "checkpoint" directly affected by the current deposit happening now.  It is still a bit hard to think about edge cases since checkpoint is still changing in size with each call which was not the case before.  My intuition is that with a queryInterval that is small like 1-2% this still works as intended but with a large enough query interval we could have unexpected behavior.  I'll work out some numerical examples

Reply:
yeah we were just discussing the same thing in a thread on slack (I'll tag you).
TLDR, is that if the interval is small (which it will be), the checkpoint does not move much and is technically more "correct" as the TVL has changed